### PR TITLE
Custom html projects page

### PR DESCRIPTION
**Pull request by nboisteault**
Fix https://github.com/3liz/lizmap-web-client/issues/1347

### CODE REVIEW

**Comment by rldhont** [comment]
If it is a new feature it has to be merged in master ?

**Comment by rldhont** [comment]
@laurentj can we create a branch release_3_4 ?

File: lizmap/modules/admin/controllers/landing_page_content.classic.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+     * Display a wysiwyg editor.
+     */
+    public function index()
+    {
+        $rep = $this->getResponse('html');
+
+        // Get HTML content from ini file
+        $lizmapConfigIniFile = jApp::configPath('lizmapConfig.ini.php');
+        $iniModifier = new jIniFileModifier ($lizmapConfigIniFile);
+
```
Comment (laurentj) [inline lizmap/modules/admin/controllers/landing_page_content.classic.php]: no, you must not use jIniFileModifier to read an ini file. It takes too much resources.

And you should not read lizmapConfig.ini.php directly, use the class LizmapService

File: lizmap/modules/admin/controllers/landing_page_content.classic.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        // Remove line breaks, spaces between > and < then transform with htmlentities to get a single string
+        $HTMLContent = htmlentities(preg_replace('/>\s+</','><',str_replace(array("\r", "\n"), '', $form->getData('HTMLContent'))));
+
+        // Save the data
+        $lizmapConfigIniFile = jApp::configPath('lizmapConfig.ini.php');
+        $iniModifier = new jIniFileModifier ($lizmapConfigIniFile);
+
+        $iniModifier->setValue("html", $HTMLContent, "landing_page_content");
```
Comment (laurentj) [inline lizmap/modules/admin/controllers/landing_page_content.classic.php]: I'm not sure lizmapConfig.ini.php is the right place to store some HTML content...

File: lizmap/www/js/filter.js [hunk 3]
```diff
@@ -538,7 +538,7 @@ var lizLayerFilterTool = function() {
                         var cval = clist[i];
                         filter+= sep + '"' + field + '"' + " " + lk + " '%" + cval + "%' ";
                         // if postgresql use ILIKE instead for WMS filtered requests
-                        sep = ' OR ';
+                        sep = ' AND ';
```
Comment (laurentj) [inline lizmap/www/js/filter.js]: Is this change really part of the new feature?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -69,6 +69,7 @@ deploy_stable:
   script:
     - make deploy_download_stable
     - make saas_release
+    - make trigger_ci
```
Comment (laurentj) [inline .gitlab-ci.yml]: Is this change really part of the new feature? Please rebase your branch

**Comment by nboisteault** [comment]
> @laurentj can we create a branch release_3_4 ?

@laurentj Can we ?

File: lizmap/www/js/filter.js (hunk 3 above)
Comment (nboisteault) [reply lizmap/www/js/filter.js]: It is not. I changed branch release_3_3 to master.

**Comment by nboisteault** [comment]
@laurentj  @mdouchin got an issue : 
`Impossible d'écrire le fichier /var/www/LIZMAP/release_3_3/lizmap/modules/view/templates/landing_page_content.tpl, assurez-vous que le répertoire /var/www/LIZMAP/release_3_3/lizmap/modules/view/templates autorise l'écriture	/var/www/LIZMAP/release_3_3/lib/jelix/utils/jFile.class.php`

He suggests to write template in `/lizmap/var/lizmap-theme-config/`. Is it ok for you ?

**Comment by mdouchin** [comment]
I propose to use a file `landing_page_content.tpl` in `/lizmap/var/lizmap-theme-config/` and use virtual templates 

```php
$templatePath = jApp::varPath('lizmap-theme-config/landing_page_content.tpl');
$templateContent = jFile::read($templatePath);
```
Then use virtual template like in https://docs.jelix.org/fr/manuel-1.6/classes-utiles/japp#autres-methodes-de-chemins

**Comment by nboisteault** [comment]
@mdouchin There is no 'source' option in CKEditor 5 for [some good reasons](https://stackoverflow.com/a/47775840/2000654). Do we agree with this answer or I try another wysiwyg widget ?

**Comment by mdouchin** [comment]
Another option would be to let the user choose one of this solution

* to create and edit HTML in the CKEditor -> for normal users
* to upload an HTML file already containing the source code, which will let power users do what they want with no editor limitation

You cannot use both at the same time

**Comment by nboisteault** [comment]
@mdouchin where do power user save his HTML file in second case ?

**Comment by nboisteault** [comment]
@mdouchin @laurentj I have rebased commits in a single one to have a more clear history. User can use wysiwyg or edit html file created to add an iframe for instance. In this case he should not use wysiwyg after as he will lose iframe. It has to be documented to warn of this.

**Comment by nboisteault** [comment]
@rldhont @laurentj  I think you can merge

**Pull request merged**